src/content/docs/2/guide/notification.mdx

```diff
@@ -0,0 +1,195 @@
+---
+title: Notifications
+description: Send native notifications to the user.
+---
+
+import PluginBreadcrumb from '@components/PluginBreadcrumb.astro';
+import { Tabs, TabItem } from '@astrojs/starlight/components';
+import CommandTabs from '@components/CommandTabs.astro';
+import Stub from '@components/Stub.astro';
+
+<PluginBreadcrumb
+	version="v2.0.0-alpha.2"
+	github="https://github.com/tauri-apps/plugins-workspace/tree/v2/plugins/notification"
+	npm="https://www.npmjs.com/package/@tauri-apps/plugin-notification"
+	cratesio="https://crates.io/crates/tauri-plugin-notification"
+	jsApi="/2/reference/js/notification"
+	docsrs="https://docs.rs/tauri-plugin-notification/2.0.0-alpha.2/tauri_plugin_notification/"
+/>
+
+Send native notifications to your user using the notification plugin.
+
+{/* TODO: Demo component */}
+
+### Security Considerations
+
+Aside from normal sanitization procedures of user input there are currently no known security considerations.
+
+## Setup
+
+First you have to install the notification plugin in the backend.
+
+<Tabs>
+    <TabItem label="Automatic">
+
+    Use the [add command](/2/reference/cli/#plugin-add) in the Tauri CLI to add the plugin:
+
+    ```bash
+    tauri plugin add notification
+    ```
+
+    {/* TODO: Does this also add it in `lib.rs`? Or does a step need to be added to cover that? */}
```

Comment:
Was just reading back through some comments on this PR, `plugin add` doesn't do this but I feel it should (really thinking of those users who don't even touch Rust, this will be a large deterrent for them).

Reply:
One option is to make the command update lib.rs for them. However, editing source code like that is no easy task, if we just do string replacing then the odds something goes wrong at some point is pretty high, and then we have users with instead of a tiny bit of Rust experience coming to use literally screaming because the black box broke and they don't know how to fix it.

One concept I've toyed with in my mind in the past is creating a tiny Typescript package that allows users to configure their backend with plugins and what not and then we generate the Rust code for them, preferably in something like `.tauri` so it's available, but not so available they'll touch it. The `.gitignore` it, update a `tauri.conf.ts` file, run Node.js on it and badda bing badda boom it generates a the source code of the project. That setup afaik wouldn't even need anything weird like a CLI update to be supported, you could literally just run `pnpm tauri dev` and it's pick up on the project in `.tauri`

Personally I feel source code editing using `plugin add` is very, very risky, and I think the kind of users that can't edit `lib.rs` are the users that need something more reliable than string replacing, so imo my Typescript config idea sounds pretty cool if I do say so myself.

Thoughts?

Reply:
I buy the complicated part, you have to dig into Rust AST which I've never even attempted. But I don't think it's risky per-se. I've seen the DX in other projects (looking at you `astro add`) where the CLI will prompt you with "here's the diff for the changes I'm going to make, does this look right and do you want to continue?".

My 2 arguments to advocate this overall are:
1. JS peeps already get overwhelmed when having to jump into Rust land in 1.0 (myself included sometimes)
2. With everything being moved into plugins for 2.0 we're going to have soooooo many more people touching these things where before it wasn't nearly as common/popular I feel

Tbh what you proposed in TypeScript sounds more complicated with a lot more hoops to jump through the logic path on rather than the CLI doing:

1. Find rust project (I think it already can do this)
2. Find the `lib.rs` file (this is either a known, static file path or can be inferred from `Cargo.toml` I assume?)
3. Find the `run()` fund inside of that (is it always called `run`? If so, then awesome!)
4. A bit of logic magic to find where to insert the plugin initialisation 

I'm sure this exists out there in a lot more places but https://docs.astro.build/en/guides/integrations-guide/svelte/#astro-add-command is what I've seen and used.

Reply:
I can put together a little package for it at some point next week probably, what I have in my head isn't as complicated as it might seem.
1. Yep
2. Static path, but not all that important
3. No need to find anything, my idea overwrites the entire source code in the `.tauri` directory. It's an approach for those that **really** don't wanna touch Rust, so they just configure it in Typescript and each time they build it the source code gets overwritten to reflect what's in the config.
4. Same answer as 3, no magic, just overwriting

Reply:
I'm open to whatever as long as we get that DX improvement (one-liner like `tauri plugin add notification`) and @lucasfernog and @wusyong give the thumbs up as I imagine that would be integrated into the core CLI.

Reply:
And just to be clear: not a blocking issue to get this merged 😅

Reply:
Put in an upstream issue for this one: https://github.com/tauri-apps/tauri/issues/7696